lib/storage/settings/settings.js

```diff
@@ -0,0 +1,307 @@
+  staticConfig: rawStaticConfig,
+  terms,
+  triggersAutomation: rawTriggersAutomation,
+  triggers,
+  urlsToPing,
+}) => {
+  return db.transaction(async (trx) => {
+    const replaceAllRows = async (tableName, newRows) => {
+      await trx(tableName).del()
+
+      const filledArray = Array.isArray(newRows) && newRows.length > 0
+      const notArrayNotEmpty = !Array.isArray(newRows) && newRows
+
+      if (filledArray || notArrayNotEmpty) {
+        await trx.insert(newRows).into(tableName)
+      }
+    }
+
+    const saveTermsInternal = saveTerms(terms, trx)
+    const saveCoins = replaceAllRows('coins', coins)
+    const saveLocales = replaceAllRows('locales', locales.locales)
+    const saveSpeedtestFiles = replaceAllRows('speedtestFiles', speedtestFiles)
+
+    const receiptOptions = Object.entries(rawReceiptOptions ?? {}).map(([field, enabled]) => ({ field, enabled }))
+    const saveReceiptOptions = replaceAllRows('receiptOptions', receiptOptions)
+    const saveOperatorInfo = replaceAllRows('operatorInfo', operatorInfo)
+
+    const triggersAutomation = Object.entries(rawTriggersAutomation).map(([triggerType, automationType]) => ({
+      triggerType,
+      automationType
+    }))
+    const saveTriggersAutomation = replaceAllRows('triggersAutomation', triggersAutomation)
+    const saveURLsToPing = replaceAllRows('urlsToPing', urlsToPing)
+
+    // static config
+    const { country, fiatCode, primaryLocale } = locales
+    const staticConfig = { ...rawStaticConfig, country, fiatCode, primaryLocale, ...machineInfo }
+    const saveStaticConfig = replaceAllRows('staticConfig', staticConfig)
+
+    const saveCustomRequest = async (customInfoRequest) => {
+      const customRequest = customInfoRequest.customRequest
+      const [input] = await trx('customInputs').insert({
+        type: customRequest.input.type,
+        constraintType: customRequest.input.constraintType,
+        label1: customRequest.input.label1,
+        label2: customRequest.input.label2
+      }).returning('rowid')
+
+      if (customRequest.input.choiceList?.length) {
+        await trx('customInputChoiceList').insert(
+          customRequest.input.choiceList.map(text => ({
+            customInput: input.rowid,
+            choiceText: text
+          }))
+        )
+      }
+
+      const [screen1] = await trx('customScreen').insert(customRequest.screen1).returning('rowid')
+      const [screen2] = await trx('customScreen').insert(customRequest.screen2).returning('rowid')
+
+      const [customRequestDb] = await trx('customRequests').insert({
+        name: customRequest.name,
+        input: input.rowid,
+        screen1: screen1.rowid,
+        screen2: screen2.rowid
+      }).returning('rowid')
+
+      return trx('customInfoRequests').insert({
+        id: customInfoRequest.id,
+        enabled: customInfoRequest.enabled,
+        customRequest: customRequestDb.rowid
+      }).returning('id')
+    }
+
+    const saveTriggers = trx('triggers').del()
```

Comment:
A bit picky, but took me a while to find this `.del()`... Maybe it could go on its own line?